client/src/components/GraphViewerComponent/GraphViewerRelationshipCreateComponent/GraphViewerRelationshipCreateComponent.js

```diff
@@ -93,56 +106,98 @@ export class GraphViewerRelationshipCreateComponent extends Component {
       document.getElementById("create-relationship-heading").focus();
     }, 200);
 
-    const { selectedNode, selectedNodes } = this.props;
-    const sourceModelId = selectedNodes.find(x => x.id !== selectedNode.id).modelId;
-    const targetModelId = selectedNode.modelId;
-
     try {
-      const relationshipItems = await new ModelService().getRelationships(sourceModelId, targetModelId);
+      const { selectedNode, selectedNodes } = this.props;
+      const source = selectedNodes.find(x => x.id !== selectedNode.id);
+      const target = selectedNode;
+      this.setState({sourceId: source.id});
+      this.setState({targetId: target.id});
+      const relationshipItems = await new ModelService().getRelationships(source.modelId, target.modelId);
+      if (relationshipItems.length === 0) {
+        this.setState({hasRelationships: false});
+      } else {
+        this.setState({hasRelationships: true});
+      }
       this.setState({ relationshipItems });
     } catch (exc) {
       this.setState({ relationshipItems: [] });
-      exc.customMessage = `Error in retrieving model. Requested ${sourceModelId}`;
+      exc.customMessage = `Error in retrieving model. Requested ${this.state.sourceModelId}`;
       eventService.publishError(exc);
     }
 
     this.setState({ isLoading: false });
   }
 
-  getNodes() {
-    const { selectedNode, selectedNodes } = this.props;
-    const source = selectedNodes && selectedNodes.find(x => x.id !== selectedNode.id);
-    const sourceId = source ? source.id : "";
-    const targetId = selectedNode ? selectedNode.id : "";
+  swap = async () => {
+    this.setState({ isLoading: true });
+    setTimeout(() => {
+      document.getElementById("create-relationship-heading").focus();
+    }, 200);
+    this.setState(({ hasSwapExecuted }) => ({ hasSwapExecuted: !hasSwapExecuted }));
 
-    return { sourceId, targetId };
+    try {
+      const { selectedNode, selectedNodes } = this.props;
+      let source = selectedNodes.find(x => x.id !== selectedNode.id);
+      let target = selectedNode;
+      if (this.state.hasSwapExecuted) {
+        source = selectedNode;
+        target = selectedNodes.find(x => x.id !== selectedNode.id);
+      }
+      this.setState({sourceId: source.id});
+      this.setState({targetId: target.id});
+      const relationshipItems = await new ModelService().getRelationships(source.modelId, target.modelId);
+      if (relationshipItems.length === 0) {
+        this.setState({hasRelationships: false});
+      } else {
+        this.setState({hasRelationships: true});
+      }
```

Comment:
See above comment re: alternative to if/else for setting state.

Reply:
Change complete